core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -204,17 +204,28 @@ private static void throwFailure(
     throw new RESTException("Unhandled error: %s", errorResponse);
   }
 
+  private String buildPath(String path) {
```

Comment:
This is looking a lot cleaner now.  However, we've now reduced this to a single call in execute where we chain `buildUri(buildPath(path)`.  This can just be collapsed to a single method since they're both private and only called in one spot.

Reply:
Made the suggested change @danielcweeks . Thank you for the review!